src/test/java/org/rulez/demokracia/pdengine/BeatTableGetPairTest.java

```diff
@@ -0,0 +1,121 @@
+package org.rulez.demokracia.pdengine;
+
+import static org.junit.Assert.*;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.rulez.demokracia.pdengine.BeatTable.Direction;
+import org.rulez.demokracia.pdengine.annotations.TestedBehaviour;
+import org.rulez.demokracia.pdengine.annotations.TestedFeature;
+import org.rulez.demokracia.pdengine.annotations.TestedOperation;
+import org.rulez.demokracia.pdengine.dataobjects.Pair;
+import org.rulez.demokracia.pdengine.exception.IsNullException;
+import org.rulez.demokracia.pdengine.exception.ReportedException;
+import org.rulez.demokracia.pdengine.testhelpers.CreatedDefaultChoice;
+
+public class BeatTableGetPairTest extends CreatedDefaultChoice {
+
+	@Before
+	public void setUp() throws ReportedException {
+		super.setUp();
+	}
+
+	@TestedFeature("Supporting functionality")
+	@TestedOperation("BeatTable")
+	@TestedBehaviour("the pair related to a and b can be obtained")
+	@Test
+	public void getPair_with_not_defined_inputs() {
+		BeatTable beatTable = new BeatTable();
+
+		assertThrows(() -> beatTable.getPair(null, null)
+				).assertMessageIs("Invalid row key");
+	}
+	
+	@TestedFeature("Supporting functionality")
+	@TestedOperation("BeatTable")
+	@TestedBehaviour("the pair related to a and b can be obtained")
+	@Test
+	public void getPair_with_not_proper_inputs() {
+		ArrayList<Choice> list = new ArrayList<Choice>();
+		Choice choice1 = new Choice("name1", "userName1");
```

Comment:
violation of TDD rule #2 http://butunclebob.com/ArticleS.UncleBob.TheThreeRulesOfTdd